### ISSUE TITLE
Text is automatically wrapped in <p> tags in nested list

### ISSUE THREAD

**Issue by A-Kun** [ckeditor/ckeditor5#16679]
## 📝 Provide detailed reproduction steps (if any)

1. Go to https://ckeditor.com/docs/ckeditor5/latest/features/source-editing.html
2. Enter source mode
3. Paste in the following markup:
```
<ol>
    <li>Item 1</li>
    <li>
        Item 2
        <ul>
            <li>
                Sub-item 1
            </li>
            <li>
                Sub-item 2
            </li>
            <li>
                Sub-item 3
            </li>
        </ul>
        Some text
    </li>
    <li>Item 3</li>
</ol>
```
5. Exit source mode, then re-enter source mode

### ✔️ Expected result

The HTML markup should be unmodified.

### ❌ Actual result

`Item 2` and `Some text` are wrapped in <p> tags.

### ❓ Possible solution

None yet. I've tried finding a workaround by wrapping the text in <span> tags but that doesn't work.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Reinmar** [comment]
Yes, CKEditor 5 does not allow to mix block and inline content on the same level (within that topmost `<li>`). This is a design decision that allowed to heavily simplified many algorithms, bringing stability to what tended to be often broken in CKEditor 4 times. 

You will need to adjust your HTML to CKEditor 5 or introduce some post-processing stripping those `<p>`s.
